src/components/Button.tsx

```diff
@@ -0,0 +1,34 @@
+import { TouchableOpacity, View } from 'react-native'
+import {
+  useRestyle,
+  spacing,
+  border,
+  backgroundColor,
+  SpacingProps,
+  BorderProps,
+  BackgroundColorProps,
+} from '@shopify/restyle'
+
+import { Text } from './Text'
+import { Theme } from '@utils/theme'
+
+const restyleFunctions = [spacing, border, backgroundColor]
+type Props = SpacingProps<Theme> &
+  BorderProps<Theme> &
+  BackgroundColorProps<Theme> & {
+    onPress: () => void
+    label: string
+    disabled: boolean
+  }
```

Comment:
You're right this is kind of a nightmare lol. Hopefully not too many more primitive components.